### DISCUSSION TITLE
IS IMPLEMENTED: Module Suggestion: todo list (daily/weekly/custom)

### DISCUSSION THREAD

**Discussion by Taschenbuch** [blish-hud/Blish-HUD#693] [Module Ideas]
A module to track your (daily and weekly) todo list 
(has been asked for a few times in the discord)

- todo list is customizable by the user. the user decides what it should show. 
- ability to add custom todo entries. 
   - e.g. "get 5 druid stones fragments daily for aurora from draconis mons vendor and hearts", 
   - e.g. "Do next daily quest for NPC xy for vision"
   - e.g. "ask guild leader if we can organize weekly meta event trains"
   - e.g. "convert volatile magic to trophy shipments"
   - e.g. "farm winter berries in bitter frost frontier for ascend trinkets"
- when a todo list entry is done, it can be checked off by the user manually via a checkbox. Or in cases where it is possible, the module could check it off automatically by using the API or other ways. 
- a todo list entry has the optional ability to uncheck the todo entry automatically at a certain time or after a certain time
  - daily server reset
  - weekly server reset
  - ...
  - custom absolute time (e.g. 8:00 pm)
  - custom relative time (e.g. in 2 hours)
    - e.g. winterberry farm and daily crafts reset not at an absolute time but after about 24hours after doing them
- maybe add some additional common todo list entries (with special support like checking the api) that the user can pick from. Examples:
  - https://fast.farming-community.eu/farming/daily-solo
  - https://gw2efficiency.com/daily
  - PSNA https://wiki.guildwars2.com/wiki/Pact_Supply_Network_Agent
    - a list of waypoints that can be copied to the clipboard for the chap would be useful
    - bonus: ability to show markers/routes for it like pathing. or point to which marker pack and which marker have to be checked for that. 
    - Note: Metalis Marker Pack in pathing module can do that already for PSNA (copy waypoints in mistlock or LA and show routes)
  - strikes, raids, dungeons (like TACO)
  - ...
- Annotations
  - The ability to check todo list entries and have customizable reset times for the todo list entries is a core feature. it should not just be a note taking module, we have blish pad module for that already. 
  - it may not need predefined todo list entries for events because event table module does that already

**Comment by Xevius95** [comment]
If the module is implemented, I would be happy if there could be a small notification for certain dailies :) E.g. WvW donor and mystic forger.

**Comment by Taschenbuch** [comment]
in development by Valorayne:
https://discord.com/channels/531175899588984842/599270434642460753/994265442384347197

**Comment by n0proxy** [comment]
I would also appreciate a built-in feature to attach a waypoint link to each to-do entry, so I can just click on a line and have it copy the wp code to my clipboard. Or click on the line and it pops open a small window where I can highlight and copy from a list of wp codes, choosing which one I want.

Support for categories would be amazing as well - ie be able to set a category called 'crafting dailies', so I can expand/collapse the category to show or hide a certain subsection of do-to entries.

Example of categories from a Wow addon below. 
![image](https://user-images.githubusercontent.com/46941718/177858350-9232ef1b-c893-4613-b8c2-c79993f9c122.png)

**Comment by Valorayne** [comment]
Thank you very much for your suggestions! 😊 
I've added them to the repo, so feel free to add more stuff that might come to mind 👍 
https://github.com/Valorayne/BlishHudTodos/discussions
